### PR TITLE
pythonPackages.pytest-order: init at 0.9.4

### PR DESCRIPTION
**Pull request by jacg**
###### Motivation for this change

Package not yet available in nixpkgs

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pytest-order/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1qd9zfpcbzm43knkg3ap22wssqabc2wn5ynlgg661xg6r6g6iy4k";
+  };
+
+  propagatedBuildInputs = [ pytest ];
+
+  meta = {
+    description = "pytest plugin that allows you to customize the order in which your tests are run";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pytest-order/default.nix:19]:
```suggestion
    description = "Pytest plugin that allows you to customize the order in which your tests are run";
```
https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

**Review by dotlambda — CHANGES_REQUESTED** [review]
"Ran 0 tests in 0.000s"

**Comment by dotlambda** [comment]
Will this be used in some other package you're about to create a pull request for? Then it would be nicer to do a single pull request for all of them.

**Review by dotlambda — CHANGES_REQUESTED** [review]
Commit message and pull request title should start with `pythonPackages`.

**Comment by jacg** [comment]
> Will this be used in some other package you're about to create a pull request for? 

No. It's used in the tests of some projects I've come across, but those are unlikely to be submitted to `nixpkgs`.

**Comment by dotlambda** [comment]
> "Ran 0 tests in 0.000s"

You need to run the tests. If they are not distributed on PyPI, you can use `fetchFromGitHub`.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110141` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.pytest-order</li>
<li>python38Packages.pytest-order</li>
<li>python39Packages.pytest-order</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.pytest-order:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.pytest-order:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.pytest-order:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>

File: pkgs/development/python-modules/pytest-order/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    repo = "pytest-order";
+    rev = "486e0966c651f3c57bbc2172e9914e478de55b6b";
+    sha256 = "1mfw2x8r5hn4g5cd6jca92lvqrd9297gsc5bvz18dzjlrm8zlycc";
+  };
+
+  propagatedBuildInputs = [
+    pytest
+    pytest-xdist
+    pytest-dependency
+    pytest-mock
```
Comment (dotlambda) [inline pkgs/development/python-modules/pytest-order/default.nix:26]: Only `pytest` should be in `propagatedBuildInputs`: https://github.com/mrbean-bremen/pytest-order/blob/v0.9.3/setup.py#L30
The rest should be put into `checkInputs`.

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110141` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>3 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.pytest-order: log <a href="https://termbin.com/j4cd">https://termbin.com/j4cd</a></li>
<li>python38Packages.pytest-order: log <a href="https://termbin.com/8ro2">https://termbin.com/8ro2</a></li>
<li>python39Packages.pytest-order: log <a href="https://termbin.com/8yoy">https://termbin.com/8yoy</a></li>
  </ul>
</details>

**Comment by jacg** [comment]
There is apparently one pending requested change here, but I cannot find any that requested changes that are unresolved.

**Comment by jacg** [comment]
After brief discussion with the maintainer of `pytest-order`, a new release (0.9.4) has been made which includes the tests. I have reverted back to using `fetchPypi` and the tests now pass.

File: pkgs/development/python-modules/pytest-order/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+buildPythonPackage rec {
+  pname = "pytest-order";
+  version = "0.9.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0b7i8z6rywnkb3skyg8bnfqgkjrwvkn64b4q07wfl1q7x65ksd26";
+  };
+
+  checkPhase = false;
```
Comment (dotlambda) [inline pkgs/development/python-modules/pytest-order/default.nix:20]:
```suggestion
```

**Pull request merged**